### DISCUSSION TITLE
Autocomple Search box

### DISCUSSION THREAD

**Discussion by Rahul-0108** [iTwin/iTwinUI-react#666] [General]
cannot find autocomplete search box , Can anyone show or it is it still not implemented ??

**Comment by mayank99** [comment]
You can use our `ComboBox` component, it is just another name for "autocomplete".

**Reply by Rahul-0108** [reply]
There is deifference between both of them
Once a dropdown value is selected , the actual state value of combox cannot be set to empty string again , 
In autocomplete text box , user can have empty value at all the time

**Reply by veekeys** [reply]
@Rahul-0108, Not sure I really understood the difference. Are you talking about multi select here? 
Maybe some example could be helpful.

**Reply by Rahul-0108** [reply]
@veekeys 
https://itwin.github.io/iTwinUI-react/?path=/story/core-combobox--basic
In this Combox , The initial value of the state value is  ''  , Now I choose India from the dropdown , Now the state value becomes 'India', Now in no way I can make the state value to '' again , (you can delete the value but onclicking outside India comes again
Autocomplete have options in the dropdown  but functions same as textbox

**Reply by veekeys** [reply]
Amm, ok. So you are missing the feature of removing the value. I think you can still do that programmatically (setting `value` to `undefined` or empty string), but you want that to happen with some click or something? Or you want the selected value to be either the typed one (if it is valid from the list) or empty one (if the typed one is invalid) ? Or you want value to be always the one, which is typed??

For us to understand better, can you please describe the workflow, where such a component would be used with the possible states? Would be really helpful to figure generic solution here.
